src/components/PriceField/PriceField.tsx

```diff
@@ -71,21 +71,20 @@ export const PriceField: React.FC<PriceFieldProps> = props => {
   );
 
   const handleChange: React.ChangeEventHandler<HTMLInputElement> = e => {
-    let newValue = e.target.value;
-    const splitCharacter = findPriceSeparator(newValue);
+    const splitCharacter = findPriceSeparator(e.target.value);
     const [integerPart, decimalPart] = e.target.value.split(splitCharacter);
 
     if (maxDecimalLength === 0 && decimalPart) {
       // this shouldn't happen - decimal character should be ignored
-      newValue = integerPart;
+      e.target.value = integerPart;
```

Comment:
This value should not be mutated. If we can't deep clone `e` with vanilla JS I suggest using lodash

Reply:
Well it works, so i guess it's okay to mutate it

From the lodash docs:

Note: This method is loosely based on the [structured clone algorithm](https://mdn.io/Structured_clone_algorithm) and supports cloning arrays, array buffers, booleans, date objects, maps, numbers, Object objects, regexes, sets, strings, symbols, and typed arrays. The own enumerable properties of arguments objects are cloned as plain objects. An empty object is returned for uncloneable values such as error objects, functions, DOM nodes, and WeakMaps.

We're talking about DOM node.